arch/x86/deflate_quick.c

```diff
@@ -201,16 +201,35 @@ static void static_emit_end_block(deflate_state *const s, int last) {
 static inline Pos quick_insert_string(deflate_state *const s, const Pos str) {
     Pos ret;
     unsigned h = 0;
+    unsigned int val;
 
-#ifdef _MSC_VER
-    h = _mm_crc32_u32(h, *(unsigned *)(s->window + str));
+#if defined(UNALIGNED_OK)
+    val = *(unsigned int *)(s->window + str);
+#else
+    memcpy(&val, (unsigned *)(s->window + str), sizeof(val)); 
```

Comment:
This is wrong... memcpy parameters should always use "unsigned char *" or "char *", never pointers to larger types. This is because compiler might implement memcpy as a macro.

Reply:
Interesting ok.

Reply:
Fixed.